tinyagentos/lazy_backend_proxy.py

```
@@ -0,0 +1,272 @@
+
+    async def _ensure_backend(self) -> None:
+        """Start the subprocess if it isn't already running."""
+        if self._proc is not None and self._proc.poll() is None:
+            return
+
+        async with self._start_lock:
+            if self._proc is not None and self._proc.poll() is None:
+                return
+
+            logger.info("lazy-proxy :%d → starting: %r", self._proxy_port, self._start_cmd)
+            self._proc = subprocess.Popen(
```

Ruff (0.15.14)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)